src/spmat/block_diagonal_matrix.py

```diff
@@ -0,0 +1,136 @@
+from typing import List, Tuple
+import numpy as np
+import scipy
+
+
+class Block:
+
+    def __init__(self, data: np.array, validate: bool = False):
+        self.data = data
+        if validate:
+            self.validate()
+
+    @property
+    def shape(self):
+        return self.data.shape
+
+    @property
```

Comment:
check out this
https://docs.python.org/3/library/functools.html#functools.cached_property

Reply:
I like this! Good to know, haven't seen cached_property before

Reply:
I come across it because of the "memoize" :))